deepchem/models/tensorgraph/tensor_graph.py

```diff
@@ -238,123 +229,146 @@ def default_generator(self,
           feed_dict[initial_state] = zero_state
         yield feed_dict
 
-  def predict_on_generator(self, generator, transformers=[]):
-    """Generates output predictions for the input samples,
-      processing the samples in a batched way.
-
-    # Arguments
-        x: the input data, as a Numpy array.
-        batch_size: integer.
-        verbose: verbosity mode, 0 or 1.
-
-    # Returns
-        A Numpy array of predictions.
-    """
-    retval = self.predict_proba_on_generator(generator, transformers)
-    if self.mode == 'classification':
-      retval = np.expand_dims(from_one_hot(retval, axis=2), axis=1)
-    return retval
-
-  def predict_proba_on_generator(self, generator, transformers=[]):
+  def predict_on_generator(self, generator, transformers=[], outputs=None):
     """
+    Parameters
+    ----------
+    generator: Generator
+      Generator that constructs feed dictionaries for TensorGraph.
+    transformers: list
+      List of dc.trans.Transformers.
+    outputs: object
+      If outputs is None, then will assume outputs = self.outputs.
+      If outputs is a Layer/Tensor, then will evaluate and return as a
+      single ndarray. If outputs is a list of Layers/Tensors, will return a list
+      of ndarrays.
     Returns:
       y_pred: numpy ndarray of shape (n_samples, n_classes*n_tasks)
     """
     if not self.built:
       self.build()
+    if outputs is None:
```

Comment:
I found that having the default behavior to set `outputs` to `self.outputs` when not specified (even when `len(self.outputs) > 1)` was quite convenient. I propose we adopt this as the default behavior instead of the old proposal.